crates/sql/src/ast/mod.rs

```diff
@@ -0,0 +1,309 @@
+
+/// Helper to override the `Display` trait for `OpQuery` enum,
+/// to convert them to their `SQL` representation.
+#[derive(Debug)]
+pub struct PrintOperator {
+    pub(crate) op: OpQuery,
+}
+
+impl fmt::Display for PrintOperator {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.op {
+            OpQuery::Cmp(op) => match op {
+                OpCmp::Eq => {
+                    write!(f, "=")
+                }
+                OpCmp::NotEq => {
+                    write!(f, "<>")
+                }
+                x => {
+                    write!(f, "{}", x)
+                }
+            },
+            OpQuery::Logic(op) => match op {
+                OpLogic::And => {
+                    write!(f, "AND")
+                }
+                OpLogic::Or => {
+                    write!(f, "OR")
+                }
+            },
+        }
+    }
+}
+
+/// Raw `SQL` value.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum SqlValue {
+    /// Boolean value
+    Bool(bool),
+    /// Hexadecimal value
+    /// ```sql
+    /// 0x0000FCAB
+    /// x'93dda0'
+    /// ```
+    Hex(String),
+    /// Unparsed number (int, float, etc.)
+    Num(String),
+    /// Unparsed long number (bigint, etc.)
+    NumLong(String),
+    /// String value
+    String(String),
+    /// Identifier (table, column, etc.)
+    ///
+    /// ```sql
+    ///  "column_name"
+    /// ```
+    Identifier(Identifier),
+    Null,
+}
+
+impl fmt::Display for SqlValue {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            SqlValue::Bool(value) => write!(f, "{}", value),
+            SqlValue::Hex(value) => write!(f, "0x{}", value),
+            SqlValue::Num(value) => write!(f, "{}", value),
+            SqlValue::NumLong(value) => write!(f, "{}", value),
+            SqlValue::Identifier(value) => write!(f, "\"{}\"", value),
+            SqlValue::String(value) => write!(f, "'{}'", value),
+            SqlValue::Null => write!(f, "NULL"),
+        }
+    }
+}
+
+/// Field expression, with optional `alias`.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, From)]
+pub enum SqlFieldExpr {
+    Name(SqlColumn),
+    Value { value: SqlValue, alias: Option<Identifier> },
+}
+
+impl SqlFieldExpr {
+    pub fn column(column: SqlColumn) -> Self {
+        SqlFieldExpr::Name(column)
+    }
+
+    pub fn value(value: SqlValue, alias: Option<Identifier>) -> Self {
+        SqlFieldExpr::Value { value, alias }
+    }
+}
+
+impl fmt::Display for SqlFieldExpr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            SqlFieldExpr::Name(column) => write!(f, "{}", column),
+            SqlFieldExpr::Value { value, alias } => match alias {
+                Some(alias) => write!(f, "{} AS {}", value, alias),
+                None => write!(f, "{}", value),
+            },
+        }
+    }
+}
+
+impl From<Ident> for SqlFieldExpr {
```

Comment:
Just to reiterate, we should remove this impl since a singular Ident shouldn't be considered a field expression.

Reply:
Done